### PR TITLE
feature: support username password auth

### PR DESCRIPTION
**Pull request by halfcrazy**
Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [x] I've tested my changes, adding unit tests if applicable.
 - [x] I've added Apache 2.0 Headers to the top of any new source files.
 - [x] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [x] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

**Comment by halfcrazy** [comment]
@jhendrixMSFT PTAL

File: autorest/adal/token.go [hunk 1]
```diff
@@ -339,6 +355,35 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
+// NewServicePrincipalTokenFromUsernamePassword create a ServicePrincipalToken from the username and password
```
Comment (mcardosos) [inline autorest/adal/token.go]: Nit, please add this link (documentation) to the comment
https://blogs.msdn.microsoft.com/wushuai/2016/09/25/resource-owner-password-credentials-grant-in-azure-ad-oauth/

**Comment by marstr** [comment]
Oh, I forgot to note in my review, because this is targeting the `master` branch, this file should be updated. https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md

**Comment by marstr** [comment]
Chatted with @vladbarosan offline, we decided on a slightly different strategy. We've got some other changes that aren't currently associated with a release tag, so we'll be submitting our own PR with the updated CHANGELOG.

**Comment by vladbarosan** [comment]
Addresses: https://github.com/Azure/go-autorest/issues/203

**Pull request merged**